### PR TITLE
MSDNet

### PR DESCRIPTION
**Pull request by cshoebridge**
Known issues:
Works, benchmarks pending, will be uploaded to overleaf doc.
Absolute memory hog, again see overleaf doc for more details.

### CODE REVIEW

File: scripts/benchmarking_scripts/msdnet_benchmarks.py [hunk 1]
```diff
@@ -1,21 +1,48 @@
...
+    y = y.cpu().numpy().squeeze()
+    vals = []
+    if x.shape[0] == 1:
+        return psnr(x, y, data_range=y.max() - y.min())
+    for i in range(x.shape[0]):
+        vals.append(psnr(x[i], y[i], data_range=y[i].max() - y[i].min()))
+    return np.array(vals).mean()
+
+
+# %%
```
Comment (AnderBiguri) [inline scripts/benchmarking_scripts/msdnet_benchmarks.py:45]: lets remove this stuff from the model folders.

**Comment by AnderBiguri** [comment]
- [ ] Lets put our MSD_net directly on the CNNs folder, rather than its own. 
- [ ] I am wondering if MSD_net should be a LIONModel itself. IF we do FBP-MSDNet, then sure, but it may be useful to have a plain vanilla pytorch MSDNet ... Need to think. 
- [ ] Lets make LION/models/post-processing/FBP_MSDNet.py where we wrap the MSDNet forward with an FDK call. I believe as for now a LIONModel can not have a LIONModel inside (bug) as it crashes at loading due to how parameters are serialized/deserialized.
- [ ] Change name of script noise2inverse2.py (at least remove the last 2)

File: LION/models/CNNs/MSDNet.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+import numpy as np
+import torch.nn as nn
+import torch
+from typing import Optional
+from LION.models.LIONmodel import LIONmodel, ModelInputType, ModelParams
+import warnings
+
+
+class MSD_Params(ModelParams):
```
Comment (AnderBiguri) [inline LION/models/CNNs/MSDNet.py:20]: Humm you'll need to explain to me how having another class just for this helps, as I don't see the advantage

Comment (cshoebridge) [reply LION/models/CNNs/MSDNet.py:20]: In N2I for example, you need to supply a model that takes an image as input, since something special needs to be done with the reconstruction step (i e the averaging stuff) for example.

Comment (cshoebridge) [reply LION/models/CNNs/MSDNet.py:20]: Rather I think it's more important to have just MSDNet rather than just FBPMSD, with the former we still have the option to do either.
